bot/cogs/filtering.py

```diff
@@ -369,6 +398,48 @@ def mod_log(self) -> ModLog:
         except discord.errors.Forbidden:
             await channel.send(f"{filtered_member.mention} {reason}")
 
+    async def _scheduled_task(self, msg: dict) -> None:
+        """Delete an offensive message once its deletion date is reached."""
+        delete_at = dateutil.parser.isoparse(msg['delete_date']).replace(tzinfo=None)
+
+        await wait_until(delete_at)
+        await self.delete_offensive_msg(msg)
+
+        self.cancel_task(msg['id'])
```

Comment:
With recent changes to the scheduler, tasks no longer need to cancel themselves. In fact, it's bad for them to do so.